src/test/java/com/commercetools/sync/sdk2/categories/utils/CategoryAssetUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,512 @@
+    assertThat(setAssetDescription).isInstanceOf(CategorySetAssetDescriptionAction.class);
+    assertThat(((CategorySetAssetDescriptionAction) setAssetDescription).getDescription())
+        .isEqualTo(newDesc);
+  }
+
+  @Test
+  void buildSetAssetTagsUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+    final List<String> oldTags = Collections.singletonList("oldTag");
+    final List<String> newTags = Collections.singletonList("newTag");
+
+    final Asset oldAsset = mock(Asset.class);
+    when(oldAsset.getTags()).thenReturn(oldTags);
+
+    final AssetDraft newAssetDraft =
+        AssetDraftBuilder.of().sources(emptyList()).name(empty()).tags(newTags).build();
+
+    final CategoryUpdateAction productUpdateAction =
+        buildSetAssetTagsUpdateAction(oldAsset, newAssetDraft).orElse(null);
+
+    assertThat(productUpdateAction).isNotNull();
+    assertThat(productUpdateAction).isInstanceOf(CategorySetAssetTagsAction.class);
+    assertThat(((CategorySetAssetTagsAction) productUpdateAction).getTags()).isEqualTo(newTags);
+  }
+
+  @Test
+  void buildSetAssetTagsUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+    final List<String> oldTags = Collections.singletonList("newTag");
+
+    final Asset oldAsset = mock(Asset.class);
+    when(oldAsset.getTags()).thenReturn(oldTags);
+
+    final AssetDraft newAssetDraft =
+        AssetDraftBuilder.of().sources(emptyList()).name(empty()).tags(oldTags).build();
+
+    final Optional<CategoryUpdateAction> productUpdateAction =
+        buildSetAssetTagsUpdateAction(oldAsset, newAssetDraft);
+
+    assertThat(productUpdateAction).isEmpty();
+  }
+
+  @Test
+  void buildSetAssetTagsUpdateAction_WithNullOldValues_ShouldBuildUpdateAction() {
+    final List<String> newTags = Collections.singletonList("newTag");
+
+    final Asset oldAsset = mock(Asset.class);
+    when(oldAsset.getTags()).thenReturn(null);
+
+    final AssetDraft newAssetDraft =
+        AssetDraftBuilder.of().sources(emptyList()).name(empty()).tags(newTags).build();
+
+    final CategoryUpdateAction productUpdateAction =
+        buildSetAssetTagsUpdateAction(oldAsset, newAssetDraft).orElse(null);
+
+    assertThat(productUpdateAction).isNotNull();
+    assertThat(productUpdateAction).isInstanceOf(CategorySetAssetTagsAction.class);
+    assertThat(((CategorySetAssetTagsAction) productUpdateAction).getTags()).isEqualTo(newTags);
+  }
+
+  @Test
+  void buildSetAssetSourcesUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+    final List<AssetSource> oldAssetSources =
+        singletonList(AssetSourceBuilder.of().uri("oldUri").build());
+    final List<AssetSource> newAssetSources =
+        singletonList(AssetSourceBuilder.of().uri("newUri").build());
+
+    final Asset oldAsset = mock(Asset.class);
+    when(oldAsset.getSources()).thenReturn(oldAssetSources);
+
+    final AssetDraft newAssetDraft =
+        AssetDraftBuilder.of().sources(emptyList()).name(empty()).sources(newAssetSources).build();
```

Comment:
.sources(emptyList()) is redundant

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/965/commits/892a0697417a09cb214e840a2c41ddcd6e13e658